server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java

```diff
@@ -58,6 +75,168 @@ public RequestParameters getParams() {
       return params;
    }
 
+   /**
+    * Handles a rollback request from a client.
+    */
+   TransactionResponse rollbackTransaction() {
+      checkConfigurationAndTransactionManager();
+      return finishTransaction(new RollbackTransactionDecodeContext(cache, (XidImpl) operationDecodeContext));
+   }
+
+   /**
+    * Handles a prepare request from a client
+    */
+   Response prepareTransaction() throws Exception {
+      checkConfigurationAndTransactionManager();
+
+      PrepareTransactionContext context = (PrepareTransactionContext) operationDecodeContext;
+      if (context.isEmpty()) {
+         //the client can optimize and avoid contacting the server when no data is written.
```

Comment:
Our implementation would sent it but I'm not preventing a user implementation to send an empty modification set.